src/server/actions/EventVolunteer.ts

```diff
@@ -0,0 +1,41 @@
+import { CreateEventVolunteerRequest } from '@/types/dataModel/eventVolunteer';
+import dbConnect from '@/utils/db-connect';
+import CMError, { CMErrorType } from '@/utils/cmerror';
+import EventVolunteerSchema from '@/server/models/EventVolunteer';
+import { mongo } from 'mongoose';
+
+// Check if volunteer has already been checked in by seeing if the checkin time field has been set.
+export async function checkInVolunteer(
+  createEventVolunteerRequest: CreateEventVolunteerRequest
+): Promise<string | null> {
+  try {
+    await dbConnect();
+
+    // Check if volunteer has been signed in
+    if (await EventVolunteerSchema.exists(createEventVolunteerRequest)) {
+      console.log('Volunteer already signed in.');
+      return null;
+    }
+    const res = await EventVolunteerSchema.create(createEventVolunteerRequest);
```

Comment:
So you are not creating an eventVolunteer you are just updating one. Use the mongoose function .findByIdAndUpdate()
You can find more information about this function here : https://www.geeksforgeeks.org/mongoose-findbyidandupdate-function/
This is a convenient function. you only need to set the checkInTime property.

Don't worry about checking to see if the volunteer is already signed in. Just throw an error if the .findByIdAndUpdate() doesn't work.